libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -180,11 +180,12 @@ string ABIFunctions::tupleDecoder(TypePointers const& _types, bool _fromMemory)
 
 		Whiskers templ(R"(
 			function <functionName>(headStart, dataEnd) <arrow> <valueReturnParams> {
-				if slt(sub(dataEnd, headStart), <minimumSize>) { revert(0, 0) }
+				if slt(sub(dataEnd, headStart), <minimumSize>) { <revertString> }
 				<decodeElements>
 			}
 		)");
 		templ("functionName", functionName);
+		templ("revertString", revertReasonIfDebug("ABI decoding error: tuple data too short"));
```

Comment:
Should we shorten the strings so that they are at most 32 characters long? `ABI decoding: tuple data too short`?

Reply:
All of them? Can do

Reply:
Just a suggestion.

Reply:
This specific one would have 34. We could have a rule that the message won't be longer than 32 which makes it easier, but the messages will be a lot less precise. I'm fine with either. What do you think?

Reply:
We should not lose precision, but the fact that this already is an error seems redundant.

Reply:
Fair enough

Reply:
Do you think it would make sense to also remove `tuple/array`, `calldata/memory` from the message?

Reply:
Don't strive too much to make it short, it was just about the "error" that seemed redundant.